src/main/java/org/elasticsearch/index/query/AndFilterParser.java

```diff
@@ -114,7 +116,11 @@ public Filter parse(QueryParseContext parseContext) throws IOException, QueryPar
         }
 
         // no need to cache this one
-        Filter filter = new AndFilter(filters);
+        BooleanQuery boolQuery = new BooleanQuery();
+        for (Filter filter : filters) {
+            boolQuery.add(filter, Occur.FILTER);
```

Comment:
Its true but its kind of confusing to understand i think, its like an unnecessary optimization where I think clarity is more important.

but really its not a big deal to me, I hope ultimately these go away anyway.